### DISCUSSION TITLE
v1.0.3-rc

### DISCUSSION THREAD

**Discussion by BlackthornYugen** [mongo-express/mongo-express#1369] [Announcements]
## What's Changed
* doc: fix README table by @rtritto in https://github.com/mongo-express/mongo-express/pull/1126
* chore: update mongodb-query-parser to v2.4.11 by @rtritto in https://github.com/mongo-express/mongo-express/pull/1136
* fix: integer id support by @rtritto in https://github.com/mongo-express/mongo-express/pull/1178
* build(deps-dev): bump webpack from 5.75.0 to 5.79.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1180
* Encode username and password as URI components in the connection string. by @macourteau in https://github.com/mongo-express/mongo-express/pull/1176
* build(deps-dev): bump node-html-parser from 6.1.4 to 6.1.5 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1139
* build(deps-dev): bump @babel/core from 7.20.12 to 7.21.4 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1168
* build(deps-dev): bump eslint from 8.33.0 to 8.38.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1175
* Fix total pages by @rtritto in https://github.com/mongo-express/mongo-express/pull/1184
* build(deps-dev): bump cypress from 12.5.1 to 12.13.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1195
* build(deps-dev): bump nodemon from 2.0.20 to 2.0.22 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1181
* build(deps-dev): bump mongodb-memory-server from 8.11.4 to 8.12.2 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1171
* Improve subtype of UUID id check by @rtritto in https://github.com/mongo-express/mongo-express/pull/1179
* fix: display `partialFilterExpression` as JSON in Indexes table by @morsko1 in https://github.com/mongo-express/mongo-express/pull/1169
* build(deps-dev): bump jquery from 3.6.3 to 3.7.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1193
* build(deps): bump mongodb-query-parser from 2.4.11 to 3.1.2 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1210
* build(deps): bump dotenv from 16.0.3 to 16.3.1 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1207
* build(deps-dev): bump eslint-plugin-unicorn from 45.0.2 to 47.0.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1201
* build(deps-dev): bump @babel/preset-env from 7.20.2 to 7.22.5 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1204
* chore: replace renderjson devDep from patch to github repo by @rtritto in https://github.com/mongo-express/mongo-express/pull/1146
* build(deps-dev): bump cypress from 12.13.0 to 12.17.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1219
* build(deps): bump body-parser from 1.20.1 to 1.20.2 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1212
* build(deps-dev): bump @babel/core from 7.21.4 to 7.22.8 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1217
* build(deps-dev): bump webpack from 5.79.0 to 5.88.1 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1213
* build(deps-dev): bump concurrently from 7.6.0 to 8.2.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1205
* chore!: remove Node 14 by @rtritto in https://github.com/mongo-express/mongo-express/pull/1094
* build(deps): bump commander from 2.20.3 to 11.0.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1221
* chore!: remove individual parameters by @rtritto in https://github.com/mongo-express/mongo-express/pull/957
* build(deps-dev): bump @babel/preset-env from 7.22.5 to 7.22.9 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1235
* build(deps-dev): bump eslint from 8.38.0 to 8.44.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1226
* fix: add ESM support to commander by @rtritto in https://github.com/mongo-express/mongo-express/pull/1223
* build(deps-dev): bump mongodb-memory-server from 8.12.2 to 8.13.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1227
* build(deps): bump html-entities from 2.3.3 to 2.4.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1231
* build(deps-dev): bump @babel/eslint-parser from 7.19.1 to 7.22.9 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1234
* build(deps-dev): bump babel-loader from 9.1.2 to 9.1.3 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1228
* build(deps-dev): bump eslint from 8.38.0 to 8.45.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1236
* build(deps): bump semver from 5.7.1 to 5.7.2 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1232
* fix: improve commander update by @rtritto in https://github.com/mongo-express/mongo-express/pull/1239
* docs: update commander README by @rtritto in https://github.com/mongo-express/mongo-express/pull/1240
* build(deps-dev): bump cypress from 12.17.0 to 12.17.2 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1247
* build(deps-dev): bump @babel/core from 7.22.8 to 7.22.9 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1246
* build(deps-dev): bump nodemon from 2.0.22 to 3.0.1 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1241
* build(deps-dev): bump eslint-plugin-unicorn from 47.0.0 to 48.0.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1244
* build(deps-dev): bump @cypress/code-coverage from 3.10.0 to 3.11.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1245
* build(deps-dev): bump webpack from 5.88.1 to 5.88.2 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1250
* build(deps-dev): bump eslint-plugin-unicorn from 48.0.0 to 48.0.1 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1252
* build(deps-dev): bump webpack-cli from 5.0.1 to 5.1.4 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1242
* fix: workaround for broken BinData serialization in mongodb-query-parser by @daniel-sc in https://github.com/mongo-express/mongo-express/pull/1254
* Update README.md by @yashdharmal in https://github.com/mongo-express/mongo-express/pull/1257
* build(deps-dev): bump eslint-plugin-import from 2.27.5 to 2.28.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1256
* add _FILE option for ME_CONFIG_MONGODB_URL by @drmathias in https://github.com/mongo-express/mongo-express/pull/1253
* chore!: update mongodb and bson to v5 by @rtritto in https://github.com/mongo-express/mongo-express/pull/1224
* build(deps-dev): bump eslint from 8.45.0 to 8.46.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1259
* fix: replace mongodb-query-parser patch with branch by @rtritto in https://github.com/mongo-express/mongo-express/pull/1255
* chore: replace json2csv with @json2csv/plainjs by @rtritto in https://github.com/mongo-express/mongo-express/pull/1260
* feat: add allowDiskUse to mongodb options by @rtritto in https://github.com/mongo-express/mongo-express/pull/1261
* build: add Node v20 by @rtritto in https://github.com/mongo-express/mongo-express/pull/1262
* refactor: routes/collection and utils by @rtritto in https://github.com/mongo-express/mongo-express/pull/1265
* build(deps-dev): bump cypress from 12.17.2 to 12.17.3 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1264
* build(deps-dev): bump mongodb-memory-server from 8.13.0 to 8.15.1 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1275
* build(deps-dev): bump eslint from 8.46.0 to 8.47.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1271
* build(deps-dev): bump @babel/preset-env from 7.22.9 to 7.22.10 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1267
* build(deps): bump free-swig from 1.5.2 to 1.5.3 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1270
* build(deps-dev): bump @babel/eslint-parser from 7.22.9 to 7.22.10 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1268
* build(deps-dev): bump node-html-parser from 6.1.5 to 6.1.6 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1274
* build(deps-dev): bump @babel/core from 7.22.9 to 7.22.10 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1269
* build(deps-dev): bump cypress from 12.17.3 to 12.17.4 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1273
* feat: add env variable to control documents per page for collection view by @carson0321 in https://github.com/mongo-express/mongo-express/pull/1281
* build(deps-dev): update @cypress/code-coverage by @rtritto in https://github.com/mongo-express/mongo-express/pull/1305
* fix: move build-assets.json reading to middleware function by @rtritto in https://github.com/mongo-express/mongo-express/pull/1311
* fix: add checks to ME_CONFIG_MONGODB_ENABLE_ADMIN env var to show/hide stats by @rtritto in https://github.com/mongo-express/mongo-express/pull/1313
* build(deps): update minor and patch versions of dependencies and dev dependencies by @rtritto in https://github.com/mongo-express/mongo-express/pull/1314
* build!: remove-node-16 by @rtritto in https://github.com/mongo-express/mongo-express/pull/1324
* build(deps-dev): bump cypress from 12.17.4 to 13.5.1 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1339
* build(deps-dev): bump @babel/eslint-parser from 7.22.15 to 7.23.3 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1336
* chore: replace busboy with @fastify/busboy by @rtritto in https://github.com/mongo-express/mongo-express/pull/1322
* feat: add Long type support to _id by @rtritto in https://github.com/mongo-express/mongo-express/pull/1321
* [Fix]:Fixed Typo in Readme by @JiyaGupta-cs in https://github.com/mongo-express/mongo-express/pull/1331
* build(deps): bump @babel/traverse from 7.19.6 to 7.23.2 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1315
* build(deps-dev): bump eslint from 8.51.0 to 8.53.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1333
* build: upgrade yarn to v4 by @rtritto in https://github.com/mongo-express/mongo-express/pull/1323
* build(deps-dev): bump concurrently from 8.2.0 to 8.2.2 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1316
* build(deps-dev): bump @cypress/code-coverage from 3.12.4 to 3.12.13 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1352
* chore! update mongodb to v6 by @rtritto in https://github.com/mongo-express/mongo-express/pull/1289
* build(deps-dev): bump @babel/core from 7.23.2 to 7.23.5 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1350
* build(deps-dev): bump mongodb-memory-server from 8.15.1 to 9.1.1 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1342
* build(deps-dev): bump eslint-plugin-unicorn from 48.0.1 to 49.0.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1345
* build(deps): bump express-fileupload from 1.4.1 to 1.4.3 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1348
* build(deps-dev): bump eslint-plugin-import from 2.28.1 to 2.29.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1346
* build(deps-dev): bump @babel/preset-env from 7.23.2 to 7.23.7 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1368
* build(deps-dev): bump cypress from 13.5.1 to 13.6.2 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1367
* build(deps-dev): bump node-html-parser from 6.1.10 to 6.1.12 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1366
* build(deps-dev): bump eslint from 8.53.0 to 8.56.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1364
* build(deps): bump @fastify/busboy from 2.0.0 to 2.1.0 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1358
* build(deps): bump free-swig from 1.5.3 to 1.5.4 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1356
* build(deps): bump @json2csv/plainjs from 7.0.3 to 7.0.4 by @dependabot in https://github.com/mongo-express/mongo-express/pull/1353

## New Contributors
* @macourteau made their first contribution in https://github.com/mongo-express/mongo-express/pull/1176
* @morsko1 made their first contribution in https://github.com/mongo-express/mongo-express/pull/1169
* @daniel-sc made their first contribution in https://github.com/mongo-express/mongo-express/pull/1254
* @yashdharmal made their first contribution in https://github.com/mongo-express/mongo-express/pull/1257
* @drmathias made their first contribution in https://github.com/mongo-express/mongo-express/pull/1253
* @carson0321 made their first contribution in https://github.com/mongo-express/mongo-express/pull/1281
* @JiyaGupta-cs made their first contribution in https://github.com/mongo-express/mongo-express/pull/1331

**Full Changelog**: https://github.com/mongo-express/mongo-express/compare/v1.0.2...v1.0.3-rc

<hr /><em>This discussion was created from the release <a href='https://github.com/mongo-express/mongo-express/releases/tag/v1.0.3-rc'>v1.0.3-rc</a>.</em>

**Comment by BlackthornYugen** [comment]
I'd like to get into the habit of starting a RC build shortly after each release, we can keep the RC branch close to master and make weekly updates to it. My goal would be to release monthly to the latest tag from the rc one. In the event of a critical CVE we would update ASAP.

We will need to update mongo-express-docker/generate-stackbrew-library.sh to handle RCs, I'll have a look at another official image like rabbitmq to see what they are doing.

**Comment by 1muen** [comment]
I used the master and RC, because we faced the major issue of the wrong paging #1184 in 1.0.2 and the minor that we couldnt edit the advanced filters if the browser window was only half the screen https://github.com/mongo-express/mongo-express/issues/1618 .
The important paging bug is fixed, the other is only fixed in master, it would be great if we could release the RC.
One inconvinience I faced was the added check to the ME_CONFIG_MONGODB_ENABLE_ADMIN env var to show/hide stats. https://github.com/mongo-express/mongo-express/pull/1313
Maybe the error could be catched and saved?

Another problem I faced with 1.0.3 is this: https://github.com/mongo-express/mongo-express/issues/1619

What is needed to get this released? More people testing it and giving feedback? More automated tests/releases?
